### DISCUSSION TITLE
alpine: client terminates by itself

### DISCUSSION THREAD

**Discussion by Rosika2** [netblue30/firejail#6437] [Q&A] [answered]
_Info: 
system: Linux Lite 6.2
firejail: 0.9.72
alpine: 2.25_

Hi all, :wave: 

although firejail ships with a dedicated **alpine** profile for the e-mail client I cannot get it to work, it seems. :neutral_face: 

Issuing the command `firejail alpine -passfile ~/.yourpassfile`, it first seems to work but after a few seconds the sandbox closes down, seemingly out of nowhere:

         
```         
ALPINE 2.25   MAIN MENU              Folder: INBOX              213 Messages +


          ?     HELP               -  Get help using Alpine           

          C     COMPOSE MESSAGE    -  Compose and send a message      

          I     MESSAGE INDEX      -  View messages in current folder 

          L     FOLDER LIST        -  Select a folder to view         

          A     ADDRESS BOOK       -  Update address book             

          S     SETUP              -  Configure Alpine Options        

          Q     QUIT               -  Leave the Alpine program        




                      For Copyright information press "?"
rosika@rosika-Lenovo-H520e ~> 
? Help                     P PrevCmd                 R RelNotes                 
O OTHER CMDS > [ListFldrs] N NextCmd                 K KBLock   
```

As you see I get my command prompt back in the third line from the bottom.

Everything´s set up correctly in alpine and the command `alpine -passfile ~/.yourpassfile` works perfectly. No problems there.

Even If I provide the ´--noprofile` option to the sandbox (for testing purposes) I arrive at the same results. The sandbox just shuts down after a few seconds. :disappointed: 

I don´t know what´s amiss and how to proceed. Can anybody help :question: 

Thank you in advance and many greetings from Rosika :slightly_smiling_face:

**Comment by rusty-snake** [comment, marked as answer]
> Even If I provide the ´--noprofile` option to the sandbox (for testing purposes) I arrive at the same results. The sandbox just shuts down after a few seconds. 😞

Try `--profile=noprofile`.

---

https://github.com/netblue30/firejail/issues/2747

https://github.com/netblue30/firejail/blob/16eefd9db8b8fe9e28ff57b41177f62d84c11125/etc/profile-a-l/alpine.profile#L10-L11

**Reply by Rosika2** [reply]
Hi @rusty-snake , :wave: 

thanks a lot for your quick reply. :heart: 

I hadn´t realized there was a question relating to the same problem before. Sorry.

> Try --profile=noprofile.

Curiously enough the sandbox shuts down just the same. Like with `--noprofile`.

**But:** the second command (referring to the workaround) worked like a charm.
I modified it a bit to fit my original command:

`firejail --private-bin=sh --include='${CFG}/allow-bin-sh.inc' --profile=alpine sh -c '(alpine -passfile ~/.yourpassfile)'`

Great. :+1: 

Now the sandbox doesn´t shut down anymore. Thanks a lot.

Could I also modify the command so that it works with a defined working directory?
I´d like to add 

`--private=/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/eigenes_alpine/Info/alpine/`

as the path to the working directory.

Many greetings from Rosika :slightly_smiling_face:

**Reply by kmk3** [reply]
> Could I also modify the command so that it works with a defined working
> directory? I´d like to add
> 
> `--private=/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/eigenes_alpine/Info/alpine/`
> 
> as the path to the working directory.

Sure, does it work or not?

**Reply by Rosika2** [reply]
@kmk3 

Hi Kelvin, :wave: 

thanks for your reply.

I haven´t tried it yet.
Until now I ran `alpine` in a normal way. I guess the respective files must be somewhere in my home directory.
I´d like to move them to the path `/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/eigenes_alpine/Info/alpine/` so that I can try whether `alpine` works with firejail using a defined working directory.

But I´m not quite sure where to find the relevant alpine data in my home directory. :thinking: 

Many greetings from Rosika :slightly_smiling_face:

**Comment by Rosika2** [comment]
Hi @kmk3, @rusty-snake and @all, :wave: 

In the meantime I tried @rusty-snake ´s workaround together with a defined work direcory. 
To this end I used the command

`firejail --private=/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/eigenes_alpine/Info/alpine --private-bin=sh --include='${CFG}/allow-bin-sh.inc' --profile=alpine sh -c '(alpine -passfile ~/.yourpassfile)'`

It worked splendidly. :+1: 

First I had to copy these already existing files and folders

- .alpine-smime (folder)
- mail (folder)
- .pinerc (file)

from my home folder to the working directory.
Works great now.

Thanks to all of you for your kind help. :heart: 

Many greetings from Rosika :slightly_smiling_face:
